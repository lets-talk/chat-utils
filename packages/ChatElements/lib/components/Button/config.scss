@import "../../common/styles/variables";
@import "../../common/styles/themes/themify";

$button-height: 40px;
$button-border-radius: 20px;
$button-box-shadow: none;
