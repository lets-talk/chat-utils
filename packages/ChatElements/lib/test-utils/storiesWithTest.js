import { withTests } from '@storybook/addon-jest';
import jestTestResults from '../../tests-results/jest-test-results.json';

export default withTests(jestTestResults, {
  filesExt: '__tests__/**.js',
});
