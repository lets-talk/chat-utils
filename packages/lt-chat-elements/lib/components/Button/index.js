import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import './index.scss';

const DELAY = 300;
const LOADER_DELAY = 900;

class Button extends Component {
  constructor(props) {
    super(props);
    this.state = {
      disabled: false,
      showLoading: false,
    };
  }

  onClick(event) {
    const { clickHandler } = this.props;

    event.preventDefault();
    event.stopPropagation();

    this.setState({ disabled: true });

    const handler = clickHandler();
    if (handler && handler instanceof Promise) {
      const loader = setTimeout(() => this.setState({ showLoading: true }), LOADER_DELAY);
      handler.then(() => {
        clearTimeout(loader);

        setTimeout(() => this.setState({
          disabled: false,
          showLoading: false,
        }), DELAY);
      });
    } else {
      this.setState({ disabled: false });
    }
  }

  render() {
    const {
      type, className, disabled, value,
    } = this.props;

    return (
      <div className="letstalk-button-container">
        <input
          type={type}
          className={classNames('letstalk-button', className)}
          value={this.state.showLoading === true ? 'Processing...' : value}
          disabled={disabled === true ? disabled : this.state.disabled}
          onClick={(event) => this.onClick(event)}
        />
      </div>
    );
  }
}

Button.propTypes = {
  type: PropTypes.oneOf(['button', 'submit', 'reset']).isRequired,
  className: PropTypes.string,
  disabled: PropTypes.bool,
  value: PropTypes.string.isRequired,
  clickHandler: PropTypes.func.isRequired,
};

export default Button;
