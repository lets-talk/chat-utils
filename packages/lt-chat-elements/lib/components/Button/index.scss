@import './config';

.letstalk-button-container {
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 10px;
}

.letstalk-button {
  position: relative;
  @include themify($themes) {
    color: themed('color-text');
    background-color: themed('color-background-primary');
  }
  height: $button-height;
  border-radius: $button-border-radius;
  box-shadow: $button-box-shadow;
  padding: 10px 25px;
  float: left;
}
