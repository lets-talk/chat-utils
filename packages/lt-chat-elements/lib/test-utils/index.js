// test-utils/index.js

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';
import { ThemeProvider } from 'styled-components';

const defaultTheme = {
  palette: {
    common: {
      black: 'rgb(0, 0, 0)',
      white: 'rgb(255, 255, 255)',
      online: 'rgb(67, 160, 71)',
      offline: 'rgb(229, 57, 53)',
      sleeping: 'rgb(253, 216, 53)',
    },
    colors: {
      primary: {
        base: 'rgb(4, 38, 73)',
        light: 'rgb(86, 165, 246)',
        dark: 'rgb(0, 0, 0)',
      },
      accent: {
        base: 'rgb(255, 87, 50)',
        light: 'rgb(255, 171, 152)',
        dark: 'rgb(203, 37, 0)',
      },
      foreground: {
        base: 'rgb(94, 124, 139)',
        light: 'rgb(151, 173, 184)',
        dark: 'rgb(53, 70, 78)',
      },
      background: {
        base: 'rgb(255, 255, 255)',
        light: 'rgb(255, 255, 255)',
        dark: 'rgb(204, 204, 204)',
      },
    },
    contrastColors: {
      dark: {
        primary: 'rgb(255, 255, 255)',
        secondary: 'rgba(255, 255, 255, 0.7)',
        disabled: 'rgba(255, 255, 255, 0.5)',
        hint: 'rgba(255, 255, 255, 0.12)',
      },
      light: {
        primary: 'rgba(0, 0, 0, 0.87)',
        secondary: 'rgba(0, 0, 0, 0.54)',
        disabled: 'rgba(0, 0, 0, 0.38)',
        hint: 'rgba(0, 0, 0, 0.12)',
      },
    },
    error: {
      light: '#e57373',
      main: '#f44336',
      dark: '#d32f2f',
      contrastText: '#fff',
    },
    divider: 'rgba(255, 255, 255, 0.1)',
    action: {
      active: 'rgba(255, 255, 255, 0.9)',
      hover: 'rgba(255, 255, 255, 0.08)',
      selected: 'rgba(255, 255, 255, 0.14)',
      disabled: 'rgba(255, 255, 255, 0.26)',
      disabledBackground: 'rgba(255, 255, 255, 0.12)',
    },
  },
  shadows: {
    '2p': '0 2px 2px 0 rgba(0, 0, 0, 0.14), 0 3px 1px -2px rgba(0, 0, 0, 0.2), 0 1px 5px 0 rgba(0, 0, 0, 0.12)',
    '3p': '0 3px 4px 0 rgba(0, 0, 0, 0.14), 0 3px 3px -2px rgba(0, 0, 0, 0.2), 0 1px 8px 0 rgba(0, 0, 0, 0.12)',
    '4p': '0 4px 5px 0 rgba(0, 0, 0, 0.14), 0 1px 10px 0 rgba(0, 0, 0, 0.12), 0 2px 4px -1px rgba(0, 0, 0, 0.2)',
    '6p': '0 6px 10px 0 rgba(0, 0, 0, 0.14), 0 1px 18px 0 rgba(0, 0, 0, 0.12), 0 3px 5px -1px rgba(0, 0, 0, 0.2)',
    '8p': '0 8px 10px 1px rgba(0, 0, 0, 0.14), 0 3px 14px 2px rgba(0, 0, 0, 0.12), 0 5px 5px -3px rgba(0, 0, 0, 0.2)',
    '16p': '0 16px 24px 2px rgba(0, 0, 0, 0.14), 0 6px 30px 5px rgba(0, 0, 0, 0.12), 0 8px 10px -5px rgba(0, 0, 0, 0.2)',
  },
  typography: {
    fontFamily: "'Nimbus sans-serif', 'Sans', 'Helvetica', 'Arial'",
    fontSize: 'calc(1 * 1px)',
    weights: {
      fontWeightUltraThin: 100,
      fontWeightThin: 300,
      fontWeightNormal: 300,
      fontWeightMedium: 500,
      fontWeightSemibold: 600,
      fontWeightBold: 700,
    },
    classes: {
      body: {
        fontSize: 'calc(16 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      caption: {
        fontSize: 'calc(12 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      footnote: {
        fontSize: 'calc(14 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      small: {
        fontSize: 'calc(10 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      headline: {
        fontSize: 'calc(24 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      subhead: {
        fontSize: 'calc(18 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      title: {
        fontSize: 'calc(20 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      jumbo: {
        fontSize: 'calc(44 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      mega: {
        fontSize: 'calc(56 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
    },
  },
  components: {
    conversationBox: {
      externalBackgroundColor: 'rgba(94, 124, 139, 0.1)',
      internalBackgroundColor: 'rgba(255, 213, 0, 0.2)',
      importantBackgroundColor: 'rgb(255, 246, 244)',
    },
    loader: {
      backgroundColor: 'rgb(252, 88, 48)',
    },
  },
};

const shallowWithTheme = (component) => {
  const context = shallow(<ThemeProvider theme={defaultTheme} />)
    .instance()
    .getChildContext();
  return shallow(component, {
    context,
    childContextTypes: ThemeProvider.childContextTypes,
  });
};

const mountWithTheme = (component) => {
  const context = shallow(<ThemeProvider theme={defaultTheme} />)
    .instance()
    .getChildContext();

  return mount(component, {
    context,
    childContextTypes: ThemeProvider.childContextTypes,
  });
};

const renderWithTheme = (component) => renderer.create(<ThemeProvider theme={defaultTheme}>{component}</ThemeProvider>);

const staticWithTheme = (component) => render(<ThemeProvider theme={defaultTheme}>{component}</ThemeProvider>);

export {
  defaultTheme,
  shallowWithTheme,
  mountWithTheme,
  staticWithTheme,
  renderWithTheme,
};
