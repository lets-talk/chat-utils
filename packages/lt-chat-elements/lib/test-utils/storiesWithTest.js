import { withTests } from '@storybook/addon-jest';
import jestTestResults from '../../tests-results/jest-test-results.json';

export default withTests({ results: jestTestResults }, {
  filesExt: '__tests__/**.js',
});
