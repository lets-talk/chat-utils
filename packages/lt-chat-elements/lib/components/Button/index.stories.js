import React from 'react';
// Storybook stuff
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { withKnobs, boolean, color, select, text } from '@storybook/addon-knobs';

import { WrapWithTheme, defaultTheme, themeOptions } from '../../utils/stories';
// Component to show on storybook
import { Button } from '../../../lib';

const stories = storiesOf('Button', module);
stories.addDecorator(withKnobs);

stories.addWithInfo(
  'Basic Usage - Button',
  'This is the basic usage of a Button.',
  () => (
    <div
      style={{
        position: 'fixed', bottom: 0, right: '10px', width: 'calc(100% - 20px)', height: '100%',
      }}
    >
      <WrapWithTheme
        themeName={select('Theme', themeOptions, defaultTheme)}
      >
        <Button
          type={select('Type', { button: 'button', submit: 'submit', reset: 'reset' }, 'button')}
          color={color('Color', '#7DBC37')}
          borderRadius={text('borderRadius', '5px')}
          className={text('className', '')}
          disabled={boolean('disabled', false)}
          value={text('value', 'Identificarse')}
          clickHandler={action('clickHandler')}
        />
      </WrapWithTheme>
    </div>
  )
);
