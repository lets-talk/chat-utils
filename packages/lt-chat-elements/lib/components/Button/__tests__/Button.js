import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { mountWithTheme } from '../../../test-utils';
import Button from '../index';

describe('Button component', () => {
  it('should render with minimal props', () => {
    const component = shallow(<Button type="submit" value="submit" clickHandler={(e) => e} />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render with data', () => {
    const props = {
      type: 'button',
      color: '#FEFEFE',
      className: 'my-custom-button-class',
      disabled: true,
      value: 'custom',
      clickHandler: () => null,
    };

    const component = shallow(<Button {...props} />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  describe('Button with and without clickHandler', () => {
    it('Test onClick event without Promise', () => {
      const mockCallBack = jest.fn();

      const props = {
        type: 'button',
        color: '#FEFEFE',
        className: 'my-custom-button-class',
        disabled: false,
        value: 'custom',
      };

      const button = shallow((<Button onClick={mockCallBack} {...props} />));
      button.find('.letstalk-button').simulate('click');
      expect(mockCallBack).toHaveBeenCalled();
    });
  });


  it('should mount with a Theme', () => {
    const component = mountWithTheme(<Button
      type="submit"
      value="submit"
      clickHandler={(e) => e}
    />);

    expect(component.length).toBe(1);
    expect(component).toMatchSnapshot();
  });
});
