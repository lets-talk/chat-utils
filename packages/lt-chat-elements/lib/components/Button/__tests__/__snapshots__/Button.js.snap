// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render with data 1`] = `
<styled.div
  className="my-custom-button-class"
>
  <styled.input
    className="letstalk-button my-custom-button-class"
    disabled={true}
    onClick={[Function]}
    type="button"
    value="custom"
  />
</styled.div>
`;

exports[`Button component should render with minimal props 1`] = `
<styled.div
  className="LT-Button-Container"
>
  <styled.input
    className="letstalk-button LT-Button-Container"
    disabled={false}
    onClick={[Function]}
    type="submit"
    value="submit"
  />
</styled.div>
`;
